### PR TITLE
Ensure compatibility with dbt>=1.4,<1.8

### LINKED ISSUE #117: Latest version is incompatible with dbt-core 1.5.0
```python
from dbt.config.runtime import UnsetProfileConfig

ImportError: cannot import name 'UnsetProfileConfig' from 'dbt.config.runtime' (/usr/local/airflow/.local/lib/python3.10/site-packages/dbt/config/runtime.py)
```

### PR DESCRIPTION
**Pull request by millin**
Fixes #117
Continuation of #118 

Dropped support for Python 3.7, dbt <1.4, Airflow <2.2

### CODE REVIEW

File: airflow_dbt_python/__version__.py
Comment (tomasfarias) [inline airflow_dbt_python/__version__.py:5]: Let's make this a minor version bump. Given the deprecations, a patch version bump is not enough.

File: pyproject.toml
Comment (tomasfarias) [inline pyproject.toml:3]: Here too.

**Comment by tomasfarias** [comment]
Approving PR for CI run. Will take a day or two to review. If all tests are green, I'm more than happy to merge. Amazing work on this @millin

**Review by Gunnnn — APPROVED** [review]

**Comment by millin** [comment]
@tomasfarias I've updated the branch, please run CI

**Comment by millin** [comment]
Latest version of `dulwich` is broken, excluded it

**Comment by millin** [comment]
@tomasfarias Could you approve again after excluding broken version of `dulwich`?
I ran all the tests locally with [act](https://github.com/nektos/act) and everything seems fine now.

File: airflow_dbt_python/utils/version.py
Comment (tomasfarias) [inline airflow_dbt_python/utils/version.py:4]: I'm afraid this will get too complicated to maintain (it is already). Should we consider not guaranteeing/dropping support for any dbt version older than the last X? X can be set to 3 and this way we could drop all the 1.5 checks.

Is there any reason to keep supporting dbt 1.5?

File: .github/workflows/ci.yaml
Comment (millin) [inline .github/workflows/ci.yaml:42]: 🤦‍♂️ forgot to specify airflow patch version in the exception

File: tests/operators/test_dbt_compile.py
Comment (tomasfarias) [inline tests/operators/test_dbt_compile.py:113]: nit:
```suggestion
        ) SELECT
```

Comment (tomasfarias) [reply tests/operators/test_dbt_compile.py:113]: Or maybe not, looks like this is coming from before. Wondering if it breaks anything.

File: airflow_dbt_python/utils/version.py
Comment (millin) [reply airflow_dbt_python/utils/version.py:4]: Ending support for older versions of airflow and dbt seems reasonable to me.

I don't know of any real reason to stay on dbt 1.4 or 1.5 and not upgrade to 1.6, except for possible unresolvable dependency conflicts that some users might have. But in any case, I think we should release this (at least one) version with dbt 1.5 support.

For the next release, the decision has to be made, either to drop support for some versions or to follow the [dbt-core eol schedule](https://docs.getdbt.com/docs/dbt-versions/core#latest-releases).

File: .github/workflows/ci.yaml
Comment (millin) [reply .github/workflows/ci.yaml]: Fixed

**Comment by millin** [comment]
@tomasfarias please re-run CI.
I fixed the airflow patch version for the exclusion and added the hosts to the allowed list.

**Comment by millin** [comment]
@tomasfarias All tests passed except of Airflow 2.5.3 + python 3.11, which is not supported. I ruled it out

**Comment by tomasfarias** [comment]
Amazing work @millin, thank you. I'll take some time this night to do a more in-depth review. If everything looks good I'm happy to merge & release a new version.

File: airflow_dbt_python/utils/version.py
Comment (tomasfarias) [reply airflow_dbt_python/utils/version.py:4]: Fine with releasing this with dbt 1.5 support.

For the next release I would personally only support the current active version + 1 old version, to ease the maintenance load. But we can discuss that later.

File: .github/workflows/ci.yaml
Comment (tomasfarias) [inline .github/workflows/ci.yaml:82]: Not clear to me why we need these two new arguments. The [docs](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-setup-python-on-ghes) for `setup-python` mention token only when discussing usage with GHES, which to be honest it's the first time I am learning about that.

Comment (tomasfarias) [reply .github/workflows/ci.yaml:82]: And architecture x64 is the default: https://github.com/actions/setup-python/blob/main/README.md?plain=1#L49

File: tests/hooks/test_git_hook.py
Comment (tomasfarias) [inline tests/hooks/test_git_hook.py:129]: nit: I prefer to check for the presence of the key, as the new change results in false positives if the Gitlab token ever happens to be a falsy value. Unlikely anyways, so I won't block merging on this.

**Review by tomasfarias — APPROVED** [review]
@millin Overall, looks good to me, amazing work. If you could just clarify the comment about CI `setup-python` requiring a github token we are good to go.

File: .github/workflows/ci.yaml
Comment (tomasfarias) [reply .github/workflows/ci.yaml]: I'll drop these two and see what happens, if CI is green I'll ship this.

File: airflow_dbt_python/__version__.py
Comment (tomasfarias) [inline airflow_dbt_python/__version__.py:5]: Due to the deprecations, and general instability of supporting multiple dbt versions, I think we should go with a v2.0. I can push a new PR to do that before release. Ultimately, we need a better translation layer with dbt: I've been toying with a `dbt-lib` for a while, but the incentive is currently not there. Anyways...

File: .github/workflows/ci.yaml
Comment (millin) [reply .github/workflows/ci.yaml]: These two arguments were necessary for local testing with `act`. I think they can be removed

File: tests/hooks/test_git_hook.py
Comment (millin) [reply tests/hooks/test_git_hook.py:129]: I changed this behavior as it was not working for me, empty variables are passed to CI even if they are not set

Comment (tomasfarias) [reply tests/hooks/test_git_hook.py:129]: Strange, well, anyways, I was being nitpicky.

File: .github/workflows/ci.yaml
Comment (tomasfarias) [reply .github/workflows/ci.yaml]: Looks like we are green. Let's release.

**Pull request merged**
